### DISCUSSION TITLE
How do I change/edit UI

### DISCUSSION THREAD

**Discussion by lumbar527** [endless-sky/endless-sky#9077] [Q&A] [answered]
Yes, I have already looked at the wiki page. Either I'm missing something, or it doesn't give me enough information

**Questions**
- How do you make a button set a condition to make other things show up. (the `[visible | active] if <condition>`)?
- This one is about the file: How do you edit the info panel (shown when you press `i`)? I found the top bar in the file and some other buttons but not the stuff for showing `combat rating`, player name, or `cargo attractiveness`.
- How/am I able to take conditions from other Endless Sky data files or the save file?
- Is some of this stuff defined in the code, not the data file?

**Comment by RisingLeaf** [comment]
> Is some of this stuff defined in the code, not the data file?

yes, all of the things you mentioned are done in the code, the data file just provides the layout

**Reply by lumbar527** [reply]
OK, so I can only change images and move stuff around?

**Reply by RisingLeaf** [reply, marked as answer]
yes

**Comment by mOctave** [comment]
What you _can_ do with the UI in the datafiles is move stuff around, drawing sprites in different locations, using different colours for text boxes that are placed in different places, etc. But, as Leaf said, what you want to do is impossible without modifying the code.

**Comment by Zitchas** [comment]
Yes, as RisingLeaf and mOctave have said, the interfaces.txt file is fairly limited. The majority of its use is
a) defining the co-ordinate for the interface element
b) defining the colors used by some interface elements

Which are very useful items to have accessible outside of the code, but there's still an awful lot that is either defined in the code, or is simply a pre-drawn image (see the images/ui folder for most of these) that is merely being placed at the provided coordinates.
